src/components/Todoitem/TodoItem.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useRef, useState, useEffect } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TempTodo } from '../../types/TempTodo';
+
+interface Props {
+  todo: Todo | TempTodo
+  onButtonRemove: (id: number) => void
+  loadingTodoIds: number[]
+  ToggleStatusCompleted: (id: number, status: boolean) => void
+  updateTitle: (id: number, newTitle: string) => void
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onButtonRemove,
+  loadingTodoIds,
+  ToggleStatusCompleted,
+  updateTitle,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isTodoEdition, setIsTodoEdition] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+  const isLoading = todo.id === 0 || loadingTodoIds.includes(todo.id);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isTodoEdition]);
+
+  const saveNewTitle = () => {
+    setIsTodoEdition(false);
+
+    if (title === todo.title) {
+      setIsTodoEdition(false);
+    }
```

Comment:
Needs to check also on redundant spaces, and looks like here is lacking `return` that will stop further code of the function.
```suggestion
    if (title.trim() === todo.title) {
      setIsTodoEdition(false);
      return;
    }
```